File: RESTful service/monitor_get.php
<?php
    $project_name = "ppm";
    $account_access_key = 'dcf338fbdea64092';
    $account_access_secret = '7409b39c2cc003d2fcbdc1859d003fa1';
    $request_url = 'https://api.acrcloud.com/v1/monitor-streams?project_name='.$project_name;
    $http_method = 'GET';
    $http_uri = "/v1/monitor-streams";
    $timestamp = time();
    $signature_version = '1';
    

    $string_to_sign = $http_method . "\n" .
                      $http_uri ."\n" .
                      $account_access_key . "\n" .
                      $signature_version . "\n" .
                      $timestamp;
    $signature = hash_hmac("sha1",$string_to_sign,$account_access_secret,true);
    $signature = base64_encode($signature);

    $headerArray = array();
    $headers = array(
        'access-key' => $account_access_key,
        'timestamp' => $timestamp,
        'signature-version' => '1',
        'signature' => $signature
    );
    foreach( $headers as $n => $v ) {
        $headerArr[] = $n .':' . $v;
    }

    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $request_url);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headerArr);
    $response = curl_exec($ch);
    var_dump($response);
    curl_close($ch);
?>

File: RESTful service/monitor_get.py
#!/usr/bin/env python
#-*- coding:utf-8 -*-

import sys
import os
import base64
import hmac
import hashlib
import urllib
import time
import requests

reload(sys)
sys.setdefaultencoding("utf8")

'''
This demo shows how to use the RESTful API to upload an audio file ( "data_type":"audio" ) into your bucket.
You can find account_access_key and account_access_secret in your account page.
Log into http://console.acrcloud.com -> "Account" (top right corner) -> "RESTful API Keys" -> "Create Key Pair".
Be Careful, they are different with access_key and access_secret of your project.
'''
account_access_key = "###YOUR_ACCESS_KEY###"
account_access_secret = "###YOUR_ACCESS_SECRET###"

requrl = "https://api.acrcloud.com/v1/monitor-streams"
http_method = "GET"
http_uri = "/v1/monitor-streams"
signature_version = "1"
timestamp = time.time()

string_to_sign = http_method+"\n"+http_uri+"\n"+account_access_key+"\n"+signature_version+"\n"+str(timestamp)

sign = base64.b64encode(
        hmac.new(account_access_secret, string_to_sign, digestmod=hashlib.sha1)
        .digest())

headers = {'access-key': account_access_key, 'signature-version': signature_version, 'signature': sign, 'timestamp':str(timestamp)}
params = {'project_name':"monitor_test"}

r = requests.get(requrl, params=params, headers=headers, verify=True)
r.encoding = "utf-8"
print r.text
